File: web/src/main/java/by/sorface/sso/web/config/security/jwt/Jwks.java
package by.sorface.sso.web.config.security.jwt;

import com.nimbusds.jose.jwk.RSAKey;

import java.security.KeyPair;
import java.security.interfaces.RSAPrivateKey;
import java.security.interfaces.RSAPublicKey;
import java.util.UUID;

public final class Jwks {

    public static RSAKey generateRsa() {
        final KeyPair keyPair = KeyGeneratorUtils.generateRsaKey(2048);
        final RSAPublicKey publicKey = (RSAPublicKey) keyPair.getPublic();
        final RSAPrivateKey privateKey = (RSAPrivateKey) keyPair.getPrivate();

        return new RSAKey.Builder(publicKey)
                .privateKey(privateKey)
                .keyID(UUID.randomUUID().toString())
                .build();
    }

}

File: web/src/main/kotlin/by/sorface/passport/web/security/oauth2/jwt/Jwks.kt
package by.sorface.passport.web.security.oauth2.jwt

import com.nimbusds.jose.jwk.RSAKey
import java.security.interfaces.RSAPrivateKey
import java.security.interfaces.RSAPublicKey
import java.util.*

object Jwks {

    fun generateRsa(): RSAKey {
        val keyPair = KeyGeneratorUtils.generateRsaKey(2048)
        val publicKey = keyPair.public as RSAPublicKey
        val privateKey = keyPair.private as RSAPrivateKey

        return RSAKey.Builder(publicKey)
            .privateKey(privateKey)
            .keyID(UUID.randomUUID().toString())
            .build()
    }

}
